agent/dockerclient/dockerapi/docker_client.go

```diff
@@ -1298,3 +1328,35 @@ func (dg *dockerGoClient) loadImage(ctx context.Context, reader io.Reader) error
 	_, err = client.ImageLoad(ctx, reader, false)
 	return err
 }
+
+type proxyReader struct {
+	io.ReadCloser
+	calls uint64
+}
+
+func (p *proxyReader) callCount() uint64 {
+	return atomic.LoadUint64(&p.calls)
+}
+
+func handleInactivityTimeout(reader io.ReadCloser, timeout time.Duration, cancelRequest func(), canceled *uint32) (io.ReadCloser, chan<- struct{}) {
```

Comment:
I'd like to see this pullled out into a different file with its own unit tests. Maybe add a todo?